packages/plugin-page-view-tracking-browser/src/page-view-tracking.ts

```diff
@@ -48,46 +81,48 @@ export const pageViewTrackingPlugin = (
         attributionConfig.trackPageViews = false;
       }
 
-      if (
-        typeof pageTrackingConfig.trackOn === 'undefined' ||
-        (typeof pageTrackingConfig.trackOn === 'function' && pageTrackingConfig.trackOn())
-      ) {
-        const event = createPageViewEvent();
-        event.event_properties = {
-          ...(await getCampaignParamsForPageViewEvent()),
-          ...event.event_properties,
-        };
-        instance.track(event);
+      if (shouldTrackOnPageLoad()) {
+        instance.track(await createPageViewEvent());
+      }
+
+      /* istanbul ignore next */
+      if (pageTrackingConfig.trackHistoryChanges) {
+        window.addEventListener('popstate', () => {
+          void trackHistoryPageView();
+        });
+
+        // There is no global browser listener for changes to history, so we have
+        // to modify pushState directly.
+        // https://stackoverflow.com/a/64927639
+        // eslint-disable-next-line @typescript-eslint/unbound-method
+        window.history.pushState = new Proxy(window.history.pushState, {
```

Comment:
I am curious that if a user pushes a history state and sdk receives the state, do we send the page view event twice?

Reply:
Believe popstate happens when the user uses the forward or back buttons, pushState only happens from things like <Link>s

Reply:
Ran some tests in a fresh create-react-app while working on this to make sure.